assets/src/components/AssignmentTable.js

```diff
@@ -120,170 +144,271 @@ function AssignmentTable (props) {
     }
   }, [currentWeekRow.current])
 
+  const [assignmentGroupNames, setAssignmentGroupNames] = useState([])
+  useEffect(() => {
+    const allGroupNames = assignments.map(ag => ag.assignmentGroup.name)
+    var uniqueGroupNames = [...new Set(allGroupNames)]
+    uniqueGroupNames.sort()
+    setAssignmentGroupNames(uniqueGroupNames)
+  }, [assignments])
+
+  const [assignmentFilter, setAssignmentFilter] = useState('')
+  const handleNameFilterChange = e => {
+    setAssignmentFilter(e.target.value)
+  }
```

Comment:
I think this line could be removed, and put inline on line 216:
`onChange={e => setAssignmentFilter(e.target.value)}`